### ISSUE TITLE
Ordering descendents

### ISSUE THREAD

**Issue by nareynolds** [overlookmotel/sequelize-hierarchy#32]
Hi,

I can't figure out how to order the descendents when executing `findAll({ hierarchy: true })`

I'm looking to order the data by a specific property as well. For example I'd like my hierarchy of folders to be returned in order of there name:
```
{
  name: "rootFolder"
  children: [
    { name: "A Folder"},
    { name: "B Folder"},
    { name: "C Folder"},
  ]
}
```

Is this possible?

**Comment by overlookmotel** [comment]
`findAll({ hierarchy: true, order: [ [ 'name' ] ] })` should work. Does it not?

**Comment by overlookmotel** [comment]
Hi @nareynolds. Can you please confirm whether this worked for you?

**Comment by nareynolds** [comment]
It did work, thanks.

Sorry, I don't know what I was doing before!

**Comment by overlookmotel** [comment]
Great, thanks for letting me know!

**Comment by leroydev** [comment]
Is this broken? Because these options
```
attributes: ['abbreviation', 'description', 'id', 'name', 'parentId'],
hierarchy: true,
order: [['name']]
``` 
Returns the following data for me (I stripped out the irrelevant attributes):
```
[  
   {  
      "name":"Test",
      "children":[  
         {  
            "name":"unit 9"
         },
         {  
            "name":"unit 10"
         },
         {  
            "name":"unit 11"
         },
         {  
            "name":"unit 12"
         },
         {  
            "name":"Operations Support"
         },
         {  
            "name":"Army"
         },
         {  
            "name":"EU Battalion"
         },
         {  
            "name":"Core Engineering"
         }
      ]
   }
]
```

**Comment by leroydev** [comment]
What about ordering when using this way to include hierarchical children (needed when viewing the tree starting not at the root):
```
include: [{
        model: model,
        as: 'descendents',
        hierarchy: true
}]
```
If I put the following order on it:
```order: [[{model: model, as: 'descendents'}, 'name', 'ASC']]```
It will only apply to the direct descendents, not to all descendents. Is there any way to apply this order to all descendents instead of only the direct descendents?

**Comment by overlookmotel** [comment]
@leroydev Hmm the code you posted should work fine. Could you post some example data for which it doesn't work?

**Comment by leroydev** [comment]
@overlookmotel:
This is for example a part of the tree that comes back in this way, instead of being ordered:
```
{
  "name": "Minecraft",
  "children": [{
    "name": "1st Battalion",
    "children": [{
      "name": "Guides"
    }, {
      "name": "1st Battalion Officers"
    }, {
      "name": "Training Team"
    }, {
      "name": "Event Staff"
    }]
  }]
}
```

**Comment by overlookmotel** [comment]
@leroydev OK thanks for this. I'll look into it.

**Comment by overlookmotel** [comment]
By the way, what version of Sequelize and sequelize-hierarchy are you using please?

**Comment by leroydev** [comment]
@overlookmotel version 0.7.0 of sequelize-hierarchy, version 3.23.4 of sequelize. Thanks for looking into this! :)

**Comment by overlookmotel** [comment]
@leroydev I've replicated this issue and found a fix. Just got to write some tests and I'll push to npm.

In the meantime, the fix is here if you need it urgently: https://github.com/overlookmotel/sequelize-hierarchy/commit/e733da1c4d4ed23011cac740a7ba1e5c17e1115c

**Comment by overlookmotel** [comment]
Fixed in v0.7.2, just published on npm.

This was a pretty bad bug - ordering is a common requirement. Thanks loads for finding it and bringing it to my attention.
